src/components/StateListItem.vue

```diff
@@ -93,15 +108,30 @@
   flex-col
   gap-2
   items-start
+  grow
 }
 
-.state-list-item__top-section { @apply
+.state-list-item__content-container { @apply
   flex
-  flex-col
+}
+
+.state-list-item__top-section { @apply
+  grid
   w-full
-  gap-2
-  md:flex-row
-  md:gap-4
+  gap-2;
+
+  grid-template-columns: 1fr auto;
+  grid-template-areas:
+    "name action"
+    "tags tags";
+}
```

Comment:
This looks like a more meaningful change than it is - it should prevent a couple blow out issues and allow us to scale the top section for small screens keeping the action slot in the same place but overflowing tags to the next line